scanner.go

```diff
@@ -234,12 +235,25 @@ func (s *Scanner) parseEscapeSequence() (rune, error) {
 			return 0, &ParseError{s.pos, err}
 		}
 
-		var r rune
+		var r byte
 		for _, b := range bytes {
 			r <<= 8
-			r |= rune(b)
+			r |= b
 		}
 		return r, nil
+	case '0', '1', '2', '3', '4', '5', '6', '7':
+		i := 0
+		for i < 3 && !s.isEOF(0) {
+			c := s.Input[s.pos.Offset+i]
+			if c < '0' || c > '7' {
+				break
+			}
+			i++
+		}
+		str := s.Input[s.pos.Offset : s.pos.Offset+i]
```

Comment:
Did gofmt do this? I would swear it never puts spaces around `:`...

Reply:
I did run the file through gofmt but I'm pretty sure I put those spaces there myself, so my guess is that gofmt is equally happy either way. The spaces are now gone.